HaxeManual/02-types.tex

```diff
@@ -46,27 +46,27 @@ \section{Basic Types}
 	\item \expr{1.0}, \expr{0.0}, \expr{-1.0}, \expr{1e10} for \type{Float}.
 \end{itemize}
 
-Basic types are not \tref{classes}{types-class-instance} in Haxe. They are implemented as \tref{abstract types}{types-abstract} and are tied to the compiler's internal operator-handling as described in the following sections.
+Basic types are not \tref{classes}{types-class-instance} in Haxe, instead they are implemented as \tref{abstract types}{types-abstract} and are tied to the compiler's internal operator-handling as described in the following sections.
 
 \subsection{Numeric types}
 \label{types-numeric-types}
 
 \define[Type]{Float}{define-float}{Represents a double-precision IEEE 64-bit floating point number.}
 
 \define[Type]{Int}{define-int}{Represents an integral number.}
-While every \type{Int} can be used where a \type{Float} is expected (that is, \type{Int} \emph{is assignable to} or \emph{unifies with} \type{Float}), the reverse is not true: Assigning a \type{Float} to an \type{Int} might lose precision and is not allowed implicitly.
+While every \type{Int} can be used where a \type{Float} is expected (that is, \type{Int} \emph{is assignable to} or \emph{unifies with} \type{Float}), the reverse is not true: assigning a \type{Float} to an \type{Int} might cause loss of precision and is therefore not implicitly allowed.
 
 \subsection{Overflow}
 \label{types-overflow}
 
-For performance reasons, the Haxe Compiler does not enforce any overflow behavior. The burden of checking for overflows falls to the target platform. Here are some platform specific notes on overflow behavior:
+For performance reasons, the Haxe Compiler does not enforce any overflow behavior. The burden of checking for numeric overflow falls to the target platform. Here are some platform specific notes on overflow behavior:
 
 \begin{description}
  \item[C++, Java, C\#, Neko, Flash:] 32-bit signed integers with usual overflow practices 
- \item[PHP, JS, Flash 8:] No native \emph{Int} type, loss of precision will occur if they reach their float limit (2\textsuperscript{52})
+ \item[PHP, JS, Flash 8:] No native \emph{Int} type, loss of precision will occur if a number reaches the float limit (2\textsuperscript{52})
 \end{description}
 
-Alternatively, the \emph{haxe.Int32} and \emph{haxe.Int64} classes can be used to ensure correct overflow behavior regardless of the platform at the cost of additional computations depending on the platform.
+If correct overflow behaviour is required, the \emph{haxe.Int32} and \emph{haxe.Int64} classes can be used although this comes at the cost of additional computations depending upon the platform.
```

Comment:
We should AE instead of BE.

Reply:
Oops. Try this:

"Alternatively, the Int32 and Int64 classes can be used to ensure correct overflow behavior although additional computations may be required on certain platforms."

I don't know whether 'all platforms' or 'certain platforms' is technically correct, I am new to Haxe!